server/controllers/user.controller.js

```diff
@@ -91,25 +103,41 @@ module.exports = {
           profile || res.status(404).send(messages.profileError);
           (res.status(200).send(Object.assign({}, user.dataValues, profile.dataValues)));
         })
-      .catch(error => {
-        return res.status(400).send(messages.badRequest);
-      });
+      .catch(error => res.status(400).send(messages.badRequest));
     })
-    .catch(error => {
-      return res.status(400).send(messages.badRequest);
-    });
+    .catch(error => res.status(400).send(messages.badRequest));
   },
+
   destroy(req, res) {
-    User.findById(req.params.id)
-    .then(user => {
-      if (!user) {
-        return res.status(404).send(messages.userNotFound);
-      }
-      return user
-      .destroy()
-      .then(user => res.status(204).send(user))
+    const eventId = req.params.event_id;
+
+    eventId && checkEventOwner(eventId, req.decoded.id)
+      .then(isOwner => {
+        isOwner && findGuest(req.params.user_id, eventId)
+        .then(guest => {
+          guest && guest.destroy();
+
+          User.findById(req.params.user_id)
+          .then(user => !user.is_activate && user.destroy());
+        })
+        .then(() => res.status(204).send(messages.guestDeleted))
+        .catch(error => res.status(404).send(messages.guestNotFound)) ||
+          res.status(403).send(messages.accessDenied);
+      }) ||
+
+      User.findById(req.params.id)
+      .then(user => {
+        user && user.destroy();
+        Guest.findAll({
```

Comment:
use cascade in the relationship  when you destroy user all guests with userId will be destroyed also